packages/react-instantsearch-hooks/src/hooks/__tests__/useInstantSearch.test.tsx

```diff
@@ -365,9 +365,10 @@ describe('useInstantSearch', () => {
 
     test('turns to loading and error when searching', async () => {
       const searchClient = createSearchClient({});
-      searchClient.search.mockImplementation(() =>
-        Promise.reject(new Error('API_ERROR'))
-      );
+      searchClient.search.mockImplementation(async () => {
+        await wait(100);
```

Comment:
Why do we need to wait?

Reply:
I think Jest's (or node's) implementation of timers and promises made it so that the error state happened immediately and skipped the loading state, which I wanted to assert in this test. Any time <16 ms was flaky so I added a bit more margin